zap/src/main/java/org/zaproxy/zap/users/User.java

```diff
@@ -166,16 +167,31 @@ public int getId() {
      * user.
      *
      * @param message the message
+     * @deprecated since 2.10.0
      */
+    @Deprecated
     public void processMessageToMatchUser(HttpMessage message) {
```

Comment:
It would be better to add a new method, e.g. `processMessageToMatchUserWithoutAuthenticate(HttpMessage)`, instead of deprecate this one, it would be more clear what the purpose of the new method is, also, others can call it if needed without passing an initiator.